packages/core/content-manager/server/services/entity-manager.js

```diff
@@ -158,25 +204,27 @@ module.exports = ({ strapi }) => ({
     );
 
     const data = { ...body, [PUBLISHED_AT_ATTRIBUTE]: new Date() };
+    const populateRelations = isRelationsPopulateEnabled(uid);
 
     const params = {
       data,
-      populate: getDeepPopulate(uid, { countMany: true, countOne: true }),
+      populate: populateRelations ? getDeepPopulate(uid, {}) : getCountDeepPopulate(uid),
```

Comment:
I don't see `getDeepRelationsCount` beeing called afterwards. It's not necessary there? (same for unpublish)

Reply:
it's inside  wrapWithEmit event, as the event of publish and unpublish is emitted there, we can not use the getDeepRelationsCount here.

Reply:
Oooh I understand the thing with `wrapWithEmitEvent`... Ah, that’s problematic a bit. `wrapWithEmitEvent` should not really be aware of this counting logic I think + it may need to count differently depending on what is wrapped.
What do you think of refactoring a bit the logic so `wrapWithEmitEvent` becomes just a an `emit` function that is called inside the controller? Same as it is done in the `entity-service`.

Reply:
Fair enough, it also seems to be a little bit complicate to decorate publish & unpublish with the wrapWithEmitEvent.
Refactoring it rn! :)